backend/src/main/java/com/jobwait/persistence/PostgresController.java

```diff
@@ -107,6 +107,8 @@ private Answers mergeAnswers(Answers persistedAnswers, Answers submittedAnswers)
             String answerType = answer.getType().toLowerCase();
             if (mergerMap.containsKey(answerType) && answer.getValue() != null) {
                 mergerMap.replace(answerType, answer);
+            } else if (mergerMap.containsKey(answerType) == false) {
+                mergerMap.put(answerType, answer);
```

Comment:
if the user had no answers at all ([]) this would do nothing and return an empty map, this else if solves this case.

Reply:
can you rewrite lines 103 to 111, it's kinda hard to read... I thought the solution would look something like
```
final_answers = persistedAnswers
for answer in submittedAnswers:
  if answer != null:
    final_answers[answer.type] = answer

return final_answers
```
The checks for if mergerMap contains keys confuse me. You say that persistedAnswers can be empty ([]) but how is this possible if the table contains null values? (shouldn't persistedAnswers be a list of answers with null values?)

Reply:
can we can fix this by inserting null answers for a user when they log in? Even outside of the scope of making a few lines here more readable, I think it makes sense to init a user with empty answers